torch/_inductor/codegen/triton.py

```diff
@@ -1275,6 +1280,44 @@ def final_reduction(value):
                 DeferredLine(name, f"tl.store({var} + {index}, {result_var}, {mask})")
             )
 
+    def vectorized_random(self, seed, mode):
+        prefix = [
+            tree.prefix
+            for tree in self.range_trees
+            if tree.prefix != "r" or self.inside_reduction
+        ]
+        blocks = [f"{p.upper()}BLOCK" for p in prefix]
+        block_size = " * ".join(blocks)
+        strides = ir.FlexibleLayout.contiguous_strides(
+            [
+                sympy.Symbol(
+                    self.cse.generate(
+                        self.compute,
+                        f"(({p}numel + {p.upper()}BLOCK - 1) // {p.upper()}BLOCK) * {p.upper()}BLOCK",
+                    ).name
+                )
+                for p in prefix
+            ]
+        )
+        offset = sympy_dot([sympy.Symbol(f"{p}offset") for p in prefix], strides)
+        offsets = self.cse.generate(
+            self.compute, f"({texpr(offset)}) // 4 + tl.arange(0, {block_size} // 4)"
+        )
+        vn = self.cse.newvar()
+        self.compute.splice(
+            f"""
+            r0, r1, r2, r3 = tl.{mode}4x({seed}, {offsets}.to(tl.uint32))
```

Comment:
how would this work with broadcasting, e.g. when we have `x` of size (4,4) and compute `x + torch.rand(4,1)`. In the current approach the `tl.rand` offset is computed from indexing expression and would produce same values as necessary, but here it looks like all offsets will be made unique (and I haven't seen tests for this, I could have overlooked, but tests are mostly `x + rand_like(x)`)

Reply:
Cool, can we add a test to make sure that the results are correct? We can't compare with eager, but e.g. if x is `ones` we can compare that its colums are equal

Reply:
Will do in next PR.  I refactored the stack to move this code into a followup PR and just make this one the refactors.